enhancements/insights/config-api.md

```diff
@@ -0,0 +1,245 @@
+	GatheringConfig GatheringConfig `json:"gatheringConfig,omitempty"`
+}
+
+type InsightsStatus struct {
+	// GatheringStatus provides basic information about the last Insights gathering.
+	GatheringStatus GatheringStatus `json:"gatheringStatus,omitempty"`
+	// SimpleContentAccessStatus provides basic information about the SCA controller.
+	SimpleContentAccessStatus SimpleContentAccessStatus `json:"simpleContentAccessStatus,omitempty"`
+	// ClusterTransferStatus provides basic information about the ClusterTransfer controller.
+	ClusterTransferStatus ClusterTransferStatus `json:"clusterTransferStatus,omitempty"`
+	// ReportStatus provides general Insights analysis results.
+	ReportStatus ReportStatus `json:"reportStatus,omitempty"`
+}
+
+type GatheringConfig struct {
+	// EnableObfuscation is flag enabling the global obfuscation of the IP addresses and base domain
+	// in the Insights archive data.
+	EnableObfuscation bool `json:"enableObfuscation,omitempty"`
+	// Enables user to force Insights data gathering by setting a new reason. Gathering can be forced even
+	// when all the gatherers are disabled.
+	ForceGatherReason string `json:"forceGatherReason,omitempty"`
+	// List of gatherers to be excluded from the gathering. All the gatherers can disabled by providing "all" value.
+	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	DisabledGatherers []string `json:"disabledGatherers,omitempty"`
+}
+
+type GatheringStatus struct {
+	// LastGatheringTime is the last time when Insights gathering finished.
+	LastGatheringTime time.Time `json:"lastGatheringTime,omitempty"`
+	// Reason provides last known reason of gathering. This is helpful
+	// especially when gathering was forced by user
+	Reason string `json:"reason,omitempty"`
+	// Status is waiting or running. If it is running then it means there is some gathering in progress.
+	// If it is waiting then a new gathering can be triggered immediately.
+	Status string `json:"status,omitempty"`
+	// Errors is a list of gathering errors from the last gathering.
+	Errors []string `json:"errors,omitempty"`
+	// List of active gatherers in the last gathering
+	Gatherers []string `json:"gatherers,omitempty"`
+}
+
+type ReportStatus struct {
+	// Number of active Insights healthchecks with low severity
+	LowCount int `json:"low,omitempty"`
+	// Number of active Insights healthchecks with moderate severity
+	ModerateCount int `json:"moderate,omitempty"`
+	// Number of active Insights healthchecks with important severity
+	ImportantCount int `json:"important,omitempty"`
+	// Number of active Insights healthchecks with critical severity
+	CriticalCount int `json:"critical,omitempty"`
+	// TotalCount is the count of all active Insights healthchecks
+	TotalCount int `json:"total,omitempty"`
+}
+
+type SimpleContentAccessStatus struct {
+	// LastDownloadTime is the last time when the SCA certs were downloaded.
+	LastDownloadTime time.Time `json:"lastDownloadTime,omitempty"`
```

Comment:
Are these latencies meaningful?  When would a user look at this timestamp?  It seems like you could have a `type=SimpleContentAccess` or some such entry in your ClusterOperator's `status.conditions` for reporting this, and admins would just see when it broke or that it had been healthy for a long time, without being able to resolve successful downloads within the healthy period.

Reply:
You will look at this when you want to check the last time of SCA download, but your suggestion works as well. Currently we set the new custom condition only when there's some problem. This basically suggests to have one (more general) condition that will be available all the time, if I got it correctly.

Reply:
What is the preferred approach? To provide the status in the clusteroperator conditions or to provide the status in the config API?

Reply:
I removed the `SimpleContentAccessStatus` and `ClusterTransferStatus` types and we will implement those in new clusteroperator conditions. I think it makes sence since there is the last time and status.